src/aap_eda/wsapi/consumers.py

```diff
@@ -345,6 +341,50 @@ def get_awx_token(self, message: WorkerMessage) -> tp.Optional[str]:
         awx_token = parent.awx_token
         return awx_token.token.get_secret_value() if awx_token else None
 
+    async def get_controller_info(
+        self, message: WorkerMessage
+    ) -> tp.Optional[ControllerInfo]:
+        """Get Controller Info."""
+        controller_info = await self.get_controller_info_from_aap_cred(message)
+        if controller_info:
+            return controller_info
+
+        awx_token = await self.get_awx_token(message)
+        if awx_token:
+            return ControllerInfo(
+                url=settings.EDA_CONTROLLER_URL,
+                token=awx_token,
+                ssl_verify=settings.EDA_CONTROLLER_SSL_VERIFY,
+            )
+        return None
+
+    @database_sync_to_async
+    def get_controller_info_from_aap_cred(
+        self, message: WorkerMessage
+    ) -> tp.Optional[ControllerInfo]:
+        """Get AAP Credential from Activation."""
+        rulebook_process_instance = models.RulebookProcess.objects.get(
+            id=message.activation_id,
+        )
+        parent = rulebook_process_instance.get_parent()
+        aap_credential_type = models.CredentialType.objects.get(
+            name=DefaultCredentialType.AAP
+        )
+        for eda_credential in parent.eda_credentials.all():
+            if eda_credential.credential_type.id == aap_credential_type.id:
+                inputs = yaml.safe_load(
+                    eda_credential.inputs.get_secret_value()
+                )
+                return ControllerInfo(
+                    url=inputs["host"],
+                    token=inputs.get("oauth_token", ""),
+                    ssl_verify=inputs.get("ssl_verify", "yes"),
```

Comment:
@hsong-rh The RH AAP calls the attribute verify_ssl not ssl_verify. https://github.com/ansible/eda-server/blob/963f5a0807815a05204ce9fe50c5669d60a7fe4f/src/aap_eda/core/management/commands/create_initial_data.py#L270

Reply:
https://github.com/ansible/eda-server/blob/main/src/aap_eda/wsapi/messages.py#L80
what we defined as string is also for future certificate purpose